code/datums/ai/basic_mobs/basic_subtrees/teleport_away_from_target.dm

```diff
@@ -0,0 +1,49 @@
+///behavior to activate ability to escape from target
+/datum/ai_planning_subtree/teleport_away_from_target
+	///minimum distance away from the target before we execute behavior
+	var/minimum_distance = 3
+	///the ability we will execute
+	var/ability_key
+
+/datum/ai_planning_subtree/teleport_away_from_target/SelectBehaviors(datum/ai_controller/controller, seconds_per_tick)
+	if(!controller.blackboard_key_exists(BB_BASIC_MOB_CURRENT_TARGET))
+		return
+	var/atom/target = controller.blackboard[BB_BASIC_MOB_CURRENT_TARGET]
+	var/distance_from_target = get_dist(target, controller.pawn)
+	if(distance_from_target >= minimum_distance)
+		return
+	var/datum/action/cooldown/ability = controller.blackboard[ability_key]
+	if(!ability?.IsAvailable())
+		return
+	controller.queue_behavior(/datum/ai_behavior/find_furthest_turf_from_target, BB_BASIC_MOB_CURRENT_TARGET, BB_ESCAPE_DESTINATION, minimum_distance)
+
+	if(controller.blackboard_key_exists(BB_ESCAPE_DESTINATION))
+		controller.queue_behavior(/datum/ai_behavior/targeted_mob_ability/and_clear_target, ability_key, BB_ESCAPE_DESTINATION)
+		return
```

Comment:
```suggestion
	if(controller.blackboard_key_exists(BB_ESCAPE_DESTINATION))
		controller.queue_behavior(/datum/ai_behavior/targeted_mob_ability/and_clear_target, ability_key, BB_ESCAPE_DESTINATION)
		return
	controller.queue_behavior(/datum/ai_behavior/find_furthest_turf_from_target, BB_BASIC_MOB_CURRENT_TARGET, BB_ESCAPE_DESTINATION, minimum_distance)
```

Should we swap these around?
Selecting behaviours happens before any of them executes, so this will never acquire a target and then escape to it within the same AI process tick, unless I have the system confused.

Reply:
> Should we swap these around? Selecting behaviours happens before any of them executes, so this will never acquire a target and then escape to it within the same AI process tick, unless I have the system confused.

yes just a copy paste error when I was moving the behavior from file to file